### DISCUSSION TITLE
Channels Stuck Closing

### DISCUSSION THREAD

**Discussion by boa76** [lightningnetwork/lnd#8204] [Q&A] [answered]
Lightning Channel Closure: I initiated the closure of a Lightning Network channel on my node approximately 10 days ago, but it still shows a "closing" status. Additionally, I noticed that I was charged twice for the closing transaction fees. How can I expedite the closure of this channel and prevent any further charges?
![Image 18-11-23 at 12 55 PM](https://github.com/lightningnetwork/lnd/assets/42274418/e9ea88e6-5d23-4b7c-8053-78d9d048a2f3)
![Image 18-11-23 at 12 55 PM](https://github.com/lightningnetwork/lnd/assets/42274418/04629594-5abf-4a00-a171-0e5b30a2bb06)

umbrel@umbrel:~ $ ~/umbrel/scripts/app compose lightning exec lnd lncli pendingchannels
{
    "total_limbo_balance": "87269",
    "pending_open_channels": [],
    "pending_closing_channels": [],
    "pending_force_closing_channels": [],
    "waiting_close_channels": [
        {
            "channel": {
                "remote_node_pub": "02ef503ab9f9fe3d78a83b4124c8ab9777fd08702b205034e6e8bd8d9d2cb84527",
                "channel_point": "7d79bdfd386c4c2972ed298d1393b47e931b6be1679247a4fe997414592f5a6c:0",
                "capacity": "100000",
                "local_balance": "87269",
                "remote_balance": "9508",
                "local_chan_reserve_sat": "1000",
                "remote_chan_reserve_sat": "1000",
                "initiator": "INITIATOR_LOCAL",
                "commitment_type": "ANCHORS",
                "num_forwarding_packages": "0",
                "chan_status_flags": "ChanStatusBorked|ChanStatusCommitBroadcasted|ChanStatusLocalCloseInitiator",
                "private": false,
                "memo": ""
            },
            "limbo_balance": "87269",
            "commitments": {
                "local_txid": "84f297da2ff444708aeacaf8702f2e76023a17462c803e7bee1419b64288fd1b",
                "remote_txid": "501c9607c9e1ceceb7d7c362ba85431019c65df544f9a6ae0204616d2fa050fb",
                "remote_pending_txid": "",
                "local_commit_fee_sat": "2562",
                "remote_commit_fee_sat": "2562",
                "remote_pending_commit_fee_sat": "0"
            },
            "closing_txid": "84f297da2ff444708aeacaf8702f2e76023a17462c803e7bee1419b64288fd1b"
        }
    ]

**Comment by microlancer** [comment]
You can bump the close fee. Also note, the mempool is slammed right now and on-chain TXs are costing 330 sats/vbyte. https://mempool.space/

**Comment by ziggie1984** [comment, marked as answer]
You are not charged twice, but I think umbrel does not account for RBF replacements (look at the tx which show the RBF history: https://mempool.space/tx/0b31240937c7b437babf9580f0b1cae44ce82c655a75eeba28845a048237ddbb). You can manually bump the fee of your closing tx via: `lncli wallet bumpclosefee`

**Comment by saubyk** [comment]
moving this issue to a discussion thread

**Reply by boa76** [reply]
Thanks all 🙏

**Comment by boa76** [comment]
I tried to bumpclosefee , here is the output , next step ?

umbrel@umbrel:~ $ ~/umbrel/scripts/app compose lightning exec lnd lncli wallet bumpclosefee
NAME:
   lncli wallet bumpclosefee - Bumps the fee of a channel closing transaction.

USAGE:
   lncli wallet bumpclosefee [command options] channel_point

DESCRIPTION:
   
  This command allows the fee of a channel closing transaction to be
  increased by using the child-pays-for-parent mechanism. It will instruct
  the sweeper to sweep the anchor outputs of transactions in the set
  of valid commitments for the specified channel at the requested fee
  rate or confirmation target.
  

OPTIONS:
   --conf_target value    the number of blocks that the output should be swept on-chain within (default: 0)
   --sat_per_vbyte value  a manual fee expressed in sat/vbyte that should be used when sweeping the output (default: 0)

**Reply by boa76** [reply]
umbrel@umbrel:~ $ ~/umbrel/scripts/app compose lightning exec lnd  lncli closechannel --force 7d79bdfd386c4c2972ed298d1393b47***************************************************
[lncli] rpc error: code = Unknown desc = channel is already in the process of being force closed
umbrel@umbrel:~ $

**Reply by boa76** [reply]
still the same .

**Reply by ziggie1984** [reply]
try this: `~/umbrel/scripts/app compose lightning exec lnd lncli wallet bumpclosefee --sat_per_vbyte  70 7d79bdfd386c4c2972ed298d1393b47e931b6be1679247a4fe997414592f5a6c:0 `

Tho it will cost you money, 70 sats/vbyte will be the effective feerate, you can choose whatever you want to, but keep in mind your channel has only 100k in sats, so it might not be worth bumping to high

**Reply by boa76** [reply]
noted ,thanks

**Reply by boa76** [reply]
Finally channel got closed, thanks all.

**Reply by ziggie1984** [reply]
can you verify that the closing tx of your channel is in your local mempool:

`bitcoin-cli getrawtransaction 2b6f12e70ca52813108ecf732325e73f534ce48c424fec6dcf8630289865d36e`. Because default mempool are already purging 11.2 sat/vbyte transaction. You need to increase your mempool size maybe to 500MB or so so that the closing tx is in your mempool

**Reply by ziggie1984** [reply]
only then can the bump of the closing happen.

**Reply by ElucGeek** [reply]
I'm in the exact same situation. This command looks correct and is not giving any error but nothing is happening. Did you solve it? Did you manage to bump the closing fee with this command or another?

sudo ~/umbrel/scripts/app compose lightning exec lnd lncli wallet bumpclosefee --sat_per_byte 100 txid:index

**Reply by ElucGeek** [reply]
That's super bad, so what are the option to recover these sats? I have also a significant amount locked and force close fee is 12sat/vB so chances that it pass soon are extremely low and even if it happens I would need to connect and rebroadcast before that to stay in the mempool.

Did you try an external bump fee service like mempool space offer? I think if I don't solve it after some more days I will try it, the fee are high, about 20$ + extra transaction fee to add, but for some Msats it's worth it I guess, even if there is no guarantee that it will work.

**Reply by guggero** [reply]
You can use [`chantools pullanchor`](https://github.com/lightninglabs/chantools/blob/master/doc/chantools_pullanchor.md) that is now included in the latest version: https://github.com/lightninglabs/chantools/releases/tag/v0.12.1 (for anchor channels that is, the old channel type is a bit more tricky).

**Reply by guggero** [reply]
If there is a force close transaction in the mempool, then yes, `chantools pullanchor` can help. If the channel hasn't been force closed yet, then no. Do you have an SCB file?

**Reply by guggero** [reply]
Yes, `pullanchor` will work here (you see the output with 330 sats, that's the anchor you can use to create a CPFP, which is what the `pullanchor` command does).

> I have recovered the node using that automatic backup option.

Ah, then that probably used the SCB to recover the channels. So you see the channels in `lncli pendingchannels`, either under waiting close or waiting force close? Then that's good, `lnd` will sweep the funds once you did the `pullanchor`.

**Reply by guggero** [reply]
Check the example in the doc I linked: https://github.com/lightninglabs/chantools/blob/master/doc/chantools_pullanchor.md

A sponsor input can be found by running `lncli listunspent` and choosing one that's large enough.

So something like that would work:
```
 chantools pullanchor --anchoraddr bc1qux0vcr3ya9v6007aqj9cyt892g4h7sfsg8y6nzaw23prtug02ucq03hzte --sponsorinput <txid:index> --changeaddr fromseed --feerate <desired_fee_rate_in_sat_per_vbyte_here>
```

No, it doesn't matter who initiated the channel. It only matters who initiated the force close which in this case must be the other party since you used SCB.

**Reply by guggero** [reply]
Ah, I just noticed now: `"local_balance": "0",` so it means you don't have any funds in that channel, only the remote party. So the anchor is not yours so you cannot pull it. Since there's no money in that channel I guess you can just ignore it.

**Reply by guggero** [reply]
Ah, you're right, the remote balance also shows 0 because for restored channels `lnd` doesn't know the actual balance... So I guess you have to check each channel if it has anchor outputs, try the `pullanchor` command on all of them and if you don't get the "no matching pubkeys found" error, then you do have a balance in that channel.

The other party can just bump the fees with `lncli wallet bumpclosefee`, assuming their `lnd` is in normal operation.
